packages/in_app_purchase/in_app_purchase_platform_interface/lib/src/in_app_purchase_platform.dart

```diff
@@ -0,0 +1,229 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'types/types.dart';
+
+/// The interface that implementations of in_app_purchase must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `in_app_purchase`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [InAppPurchasePlatform] methods.
+abstract class InAppPurchasePlatform extends PlatformInterface {
+  /// Constructs a InAppPurchasePlatform.
+  InAppPurchasePlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static InAppPurchasePlatform? _instance;
+
+  /// The default instance of [InAppPurchasePlatform] to use.
+  static InAppPurchasePlatform get instance {
```

Comment:
If we have this `instance`, `InAppPurchaseConnection.instance` will return a `InAppPurchasePlatform` instead of `InAppPurchaseConnection`. Do we really need this _instance here? 

I think federating this plugin fit more of the scenario described here: https://github.com/flutter/flutter/issues/43368 and we don't currently have a solution yet.

Reply:
Removed the `_instance` and `instance` getter/ setter as per [comment](https://github.com/flutter/plugins/pull/3781#discussion_r609060157).